spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindOverridableMethodCall.java

```diff
@@ -100,53 +92,32 @@ public void visit(JavaClass obj) {
         for (int i = 0; i < obj.getBootstrapMethods().length; ++i) {
-            BootstrapMethod bm = obj.getBootstrapMethods()[i];
             CallerInfo ctor = refCallerConstructors.get(i);
             CallerInfo clone = refCallerClones.get(i);
             Collection<XMethod> callers = refCalleeToCallerMap.get(i);
             if (ctor == null && clone == null && (callers == null || callers.isEmpty())) {
                 continue;
             }
-            for (int arg : bm.getBootstrapArguments()) {
-                Constant c = obj.getConstantPool().getConstant(arg);
-                if (!(c instanceof ConstantMethodHandle)) {
-                    continue;
-                }
-                ConstantMethodHandle cmh = (ConstantMethodHandle) c;
-                c = getConstantPool().getConstant(cmh.getReferenceIndex());
-                if (!(c instanceof ConstantMethodref) && !(c instanceof ConstantInterfaceMethodref)) {
-                    continue;
-                }
-                ConstantCP cp = (ConstantCP) c;
-                if (cp.getClassIndex() != getThisClass().getClassNameIndex()) {
-                    return;
-                }
-                ConstantNameAndType cnat = (ConstantNameAndType) getConstantPool()
-                        .getConstant(cp.getNameAndTypeIndex());
-                Optional<Method> metOpt = Arrays.stream(getThisClass().getMethods())
-                        .filter(m -> m.getNameIndex() == cnat.getNameIndex()
-                                && m.getSignatureIndex() == cnat.getSignatureIndex())
-                        .findAny();
-                if (!metOpt.isPresent()) {
-                    continue;
-                }
-                XMethod method = getXClass().findMethod(metOpt.get().getName(), metOpt.get().getSignature(),
-                        metOpt.get().isStatic());
-                if (ctor != null && checkDirectCase(ctor.method, method, "MC_OVERRIDABLE_METHOD_CALL_IN_CONSTRUCTOR",
-                        LOW_PRIORITY, ctor.sourceLine)) {
-                    checkAndRecordCallFromConstructor(ctor.method, method, ctor.sourceLine);
-                }
-                if (clone != null && checkDirectCase(clone.method, method, "MC_OVERRIDABLE_METHOD_CALL_IN_CLONE",
-                        NORMAL_PRIORITY, clone.sourceLine)) {
-                    checkAndRecordCallFromClone(clone.method, method, clone.sourceLine);
-                }
-                if (callers != null) {
-                    for (XMethod caller : callers) {
-                        if (method.isPrivate() || method.isFinal()) {
-                            checkAndRecordCallBetweenNonOverridableMethods(caller, method);
-                        } else {
-                            checkAndRecordCallToOverridable(caller, method);
-                        }
+            Method method = BootstrapMethodsUtil.getMethodFromBootstrap(obj, i, getConstantPool(), getThisClass());
```

Comment:
Only this class depends on this new util method, so I suggest you to convert the util method into a method in this class. It is better to reduce published API as much as possible.

Reply:
Sorry, probably GitHub tricked you (as it also tricked me) because it hides the largest file by default. Actually, I created this utility class especially because both the current detector (`FindPotentialSecurityCheckBasedOnUntrustedSource`) and `FindOverridableMethodCall` uses this method. I also expect more (already existing) detectors in the future to cope with lambdas so this will probably also use it.